lib/index.js

```diff
@@ -6,111 +6,125 @@ const { getDungeonExpWeight } = require("./dungeonExpWeight");
-        /**
-         * Gets the player's weight.
-         * @param {string} player - Either a Minecraft username or a Minecraft UUID (can be with dashes).
-         * @param {boolean} [returnUsername=false] - Should the function return the player's username.
-         * @returns {Promise<{total: number, slayer: number, skill: {overflow: number, base: number}, uuid: string, username?: string, catacombs: {completion: {base: number, master: number}, experience: number}}>}
-         */
-        getWeight: async (player, returnUsername = false) =>
-            getWeight(player.length > 16 ? player : await getUUID(player), apiKey, returnUsername),
+        if (returnUsername) {
+            if (!username)
+                username = await fetchUsername(uuid);
+            return { username, uuid, ...weight };
+        } else return { uuid, ...weight };
+    }
 
-        /**
-         * Gets the player's weight.
-         * @param {string} uuid - Player's Minecraft UUID, can be with dashes.
-         * @param {boolean} [returnUsername=false] - Should the function return the player's username.
-         * @returns {Promise<{total: number, slayer: number, skill: {overflow: number, base: number}, uuid: string, username?: string, catacombs: {completion: {base: number, master: number}, experience: number}}>}
-         */
-        getWeightFromUUID: async (uuid, returnUsername = false) =>
-            getWeight(uuid, apiKey, returnUsername),
+    /**
+     * Gets the player's weight.
+     * @param {string} username - Player's Minecraft username.
+     * @param {boolean} [returnUsername=false] - Should the function return the player's username.
+     * @returns {Promise<PlayerWeights>} the weights calculated for the player.
+     */
+    async getWeightFromUsername(username, returnUsername = false) {
+        return this.getWeight(await getUUID(username), returnUsername);
+    }
 
-        /**
-         * Gets the player's weight.
-         * @param {string} username - Player's Minecraft username.
-         * @param {boolean} [returnUsername=false] - Should the function return the player's username.
-         * @returns {Promise<{total: number, slayer: number, skill: {overflow: number, base: number}, uuid: string, username?: string, catacombs: {completion: {base: number, master: number}, experience: number}}>}
-         */
-        getWeightFromUsername: async (username, returnUsername = false) =>
-            getWeight(await getUUID(username), apiKey, returnUsername),
+    /**
+     * Gets the player's weight.
+     * @param {string} player - Either a Minecraft username or a Minecraft UUID (can be with dashes).
+     * @param {boolean} [returnUsername=false] - Should the function return the player's username.
+     * @returns {Promise<PlayerWeights>} the weights calculated for the player.
+     */
+    async getWeight (player, returnUsername = false) {
+        return this.getWeight(player.length > 16 ? player : await getUUID(player), returnUsername);
```

Comment:
This should also be a `this.getWeightFromUUID`, since that results in an infinite loop until it errors out for reaching the maximum call stack size.

Reply:
There also shouldn't be a space between the `getWeight` and `(player, returnUsername = false)`, so it matches the rest of the codebase.

Reply:
oh thanks don't know how I missed that

Reply:
resolved by 3f5694a and 39114f5